pages/form/details.tsx

```diff
@@ -37,7 +38,8 @@ const Form = () => {
 
   return (
     <form onSubmit={onSubmit}>
-      <h1 className="font-bold text-4xl mt-8 dark:text-ocf-gray mb-5">
+      <BackButton />
```

Comment:
Move this outside the form so it doesn't automatically submit the form. It's also unrelated semantically so should be separated. I know this might ruin some margins, but then perhaps the margins should be outside the form too.